bin/beepmein

```diff
@@ -0,0 +1,281 @@
+    # job was generated ok
+    atrm "$tmpjob"
+else
+    # error
+    return 1
+fi
+}
+
+# help or test
+if [ "$1" == "--help" ] || [ "$1" == "-h" ] || [ $# -eq 0 ]; then
+    help; exit
+elif [ "$1" == "--test" ]; then
+    test; exit
+fi
+
+# list all 'at' jobs, last non-empty line only
+if [ "$1" == "-l" ] || [ "$1" == "--list" ]; then
+    atq | sort -r | while read -r job rest ; do
+        echo -n "$job $rest | "
+        at -c "$job" | tac | grep -vm 1 '^[[:space:]]*$'
+    done
+    exit
+fi
+
+# --set-alarm (via softlink)
+if [ "$1" == "--set-alarm" ]; then
+    if [[ -z "$2" ]]; then
+        alarmpath="$(yad --on-top --center --title="Select alarm sample path" --file)"
+    else
+        alarmpath=$(readlink -f "$2") # must be absolute path
+    fi
+    [[ -f "$alarmpath" ]] || { superecho "No valid alarm path given." ; beep "2"; exit 1; }
+    # f=force
+    ln -fs "$alarmpath" "$config/alarm"
+    exit
+fi
+# --set-remind (via softlink)
+if [ "$1" == "--set-remind" ]; then
+    if [[ -z "$2" ]]; then
+        remindpath="$(yad --on-top --center --title="Select reminder sample path" --file)"
+    else
+        remindpath=$(readlink -f "$2") # must be absolute path
+    fi
+    [[ -f "$remindpath" ]] || { superecho "No valid reminder path given." ; beep "2"; exit 1; }
+    # f=force
+    ln -fs "$remindpath" "$config/remind"
+    exit
+fi
+
+# main
+if [ "$1" != "--reaction" ] ; then
+    if [ "$1" == "--nr" ] ; then       # --nr must be first
+        # chop off first argument
+        noreason="true" && shift
+    fi
+    if [ "${!#}" == "--alarm" ] ; then # --alarm must be last
+        # chop off last argument
+        alarm="--alarm" && set -- "${@:1:$(($#-1))}"
+        testAlarmPath
+    fi
+    # input
+    attime="$*" #concat
+    # special case for minutes
+    if [[ $attime == ?(-)+([0-9]) ]] ; then # is integer
+        attime="now + $attime minutes"
+        # lets add delay in seconds, since 'at' is only about minutes,
+        # making this shortcut the only to seconds precise format.
+        setdelay="1"
+    fi
+    
+    # validate at time format, exit if fails
+    validate "$attime" || { >&2 superecho "at time format error." ; beep "2"; exit 1; } # beep two times for error
+    
+    # user input reason
+    if [[ $noreason != true ]]; then
+        read -rp "reason? " reason || reason="$(yad --on-top --center --title="beepmein" --width=300 --separator="" --form --field="reason?")"
+    fi
+    [[ -z "$reason" ]] && reason="beep beep beep"
+    echo "beepmein: $attime, reason: $reason $alarm"
+    
+    # if custom input
+    (( setdelay )) && delay="$(date +%_S.%N)"
+    
+    # set 'at'
+    at "$attime" <<< "$(echo "sleep $delay; DISPLAY=${DISPLAY} beepmein --reaction $alarm '${reason}'")" > /dev/null 2>&1 || { superecho "at input error." ; beep "2"; exit 1; } # beep two times for error
+    
+else
+    # reaction
+    shift
+    if [ "$1" == "--alarm" ]; then
+        shift
+        notify-send "$1" &
```

Comment:
What should happen if notify-send fails? (dbus not running or whatever).

Reply:
Hopefully blink and sound would still work, or you have some other idea? There could be a yad popup?